src/input.rs

```diff
@@ -334,6 +334,7 @@ impl State {
             }
             Action::Spawn(command) => {
                 spawn(command);
+                //FIXME: move_cursor_to_focused_tile doesn't work here
```

Comment:
I guess instead of all these warps here, you could check if the active window changed somewhere around `refresh_and_flush_clients()`. That will also get spawning and closing and everything else, and remove all the code duplication.

It will also be a good place to put checks for when we don't want to move the mouse, e.g. when the screenshot UI is open.

Reply:
implemented in [5310e88](https://github.com/YaLTeR/niri/pull/245/commits/5310e88ab540d38ed71afadbe33c7ed390f70935), again was not able to test multi monitor configuration yet.

Reply:
Tested on dual monitor setup and also works as expected